packages/wallets/torus/src/adapter.ts

```diff
@@ -1,109 +1,98 @@
 import {
+    BaseMessageSignerWalletAdapter,
     BaseSignerWalletAdapter,
+    EventEmitter,
+    pollUntilReady,
+    WalletAccountError,
     WalletConnectionError,
+    WalletDisconnectedError,
     WalletDisconnectionError,
     WalletError,
-    WalletKeypairError,
     WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
     WalletSignTransactionError,
-    WalletWindowBlockedError,
     WalletWindowClosedError,
 } from '@solana/wallet-adapter-base';
-import { Keypair, PublicKey, Transaction } from '@solana/web3.js';
-import OpenLogin, { OPENLOGIN_NETWORK, OpenLoginOptions } from '@toruslabs/openlogin';
-import { getED25519Key } from '@toruslabs/openlogin-ed25519';
+import { PublicKey, Transaction, Message, Cluster } from '@solana/web3.js';
+import Torus, { TorusParams } from '@toruslabs/solana-embed';
 
 export interface TorusWalletAdapterConfig {
-    options: Partial<OpenLoginOptions> & Omit<OpenLoginOptions, 'network'>;
+    torusParams?: TorusParams;
+    // network?: WalletAdapterNetwork;
+    pollInterval?: number;
+    pollCount?: number;
 }
+interface TorusWindow extends Window {
+    torus: Torus | null;
+}
+
+declare const window: TorusWindow;
 
 export class TorusWalletAdapter extends BaseSignerWalletAdapter {
-    private _options: OpenLoginOptions;
     private _connecting: boolean;
-    private _openLogin: OpenLogin | null;
-    private _keypair: Keypair | null;
+    private _torus: Torus | null;
+    private _publicKey: PublicKey | null;
+    private _config: TorusParams;
 
     constructor(config: TorusWalletAdapterConfig) {
         super();
-        this._options = { uxMode: 'popup', network: OPENLOGIN_NETWORK.MAINNET, ...config.options };
         this._connecting = false;
-        this._openLogin = null;
-        this._keypair = null;
+        this._torus = null;
+        this._publicKey = null;
+        this._config = config.torusParams || {};
+        // if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+        // this.ready = true;
     }
 
     get publicKey(): PublicKey | null {
-        return this._keypair?.publicKey || null;
+        return this._publicKey;
     }
 
     get ready(): boolean {
-        return typeof window !== 'undefined';
+        return true
     }
 
     get connecting(): boolean {
         return this._connecting;
     }
 
     get connected(): boolean {
-        return !!this._keypair;
+        return !!this._torus?.isLoggedIn;
     }
 
     async connect(): Promise<void> {
         try {
             if (this.connected || this.connecting) return;
             this._connecting = true;
 
-            let openLogin: OpenLogin;
-            let privateKey: string;
+            // Check if torus is initialized, torus.init({config})
+            let torus = typeof window !== 'undefined' && window.torus;
+            if (!torus) torus = new Torus();
+            window.torus = torus;
```

Comment:
This probably shouldn't happen in practice, but this code will error if called when `window` is undefined because it tries to set `window.torus` in either case. It should throw `WalletNotReadyError` instead of proceeding.